mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -0,0 +1,221 @@
+import gorilla
+import logging
+import mlflow.pytorch
+import os
+import pytorch_lightning as pl
+import shutil
+import tempfile
+from pytorch_lightning.core.memory import ModelSummary
+from pytorch_lightning.utilities import rank_zero_only
+from mlflow.utils.autologging_utils import try_mlflow_log, wrap_patch
+
+logging.basicConfig(level=logging.ERROR)
+
+every_n_epoch = 1
+
+
+# autolog module uses `try_mlflow_log` - mlflow utility to log param/metrics/artifacts into mlflow
+# MlflowLogger(Pytorch Lightning's inbuild class),
+# following are the downsides in using MlflowLogger.
+# 1. MlflowLogger doesn't provide a mechanism to store an entire model into mlflow.
+#    Only model checkpoint is saved.
+# 2. For storing the model into mlflow `mlflow.pytorch` library is used
+# and the library expects `mlflow` object to be instantiated.
+# In case of MlflowLogger, Run management is completely controlled by the class and
+# hence mlflow object needs to be reinstantiated by setting
+# tracking uri, experiment_id and run_id which may lead to a race condition.
```

Comment:
@shrinath-suresh could we remove this comment in favor of filing an issue against PytorchLightning & linking that? I worry that we may lose track of these details / not take action otherwise :)